### DISCUSSION TITLE
`public` 폴더 만들기 건의

### DISCUSSION THREAD

**Discussion by WaiNaat** [woowacourse-teams/2023-pium#27] [뜨거운 토론]
안녕하세요! 친애하는 프론트엔드 크루 여러분🍱

### 안건

`index.html`과 `mockServiceWorker.js`를 `public/` 안에 담아볼까요?

### 설명

우선 현재 webpack config를 건드려야 해요. 개발 서버에서  `mockServiceWorker.js`에 접근할 필요가 있기 때문입니다.
![image](https://github.com/woowacourse-teams/2023-pium/assets/77872742/89771f92-db2d-4593-b19d-3263313d73e2)
이런 느낌으로 static directory 설정을 해야 해요.

근데 이렇게 설정하면 문제가 생깁니다.

> static directory: 개발 서버에서 주소를 통해 서버 내 파일을 요청할 경우 어디서부터 찾아야 하는지를 명시함

![src접근](https://github.com/woowacourse-teams/2023-pium/assets/77872742/edddad60-3bcc-4592-900e-5669949bc834)

저희가 최상단 디렉토리를 static으로 지정하기 때문에 이런 식으로 소스 코드 폴더에까지 접근이 가능해요.
`index.html`과 `mockServiceWorker.js`를 `public/` 안에 담으면 `localhost:8282/../src` 이런 식으로 접근이 불가능하기 때문에 코드를 지킬 수 있어요.

**Comment by WaiNaat** [comment]
추가 설명 들어갑니다

### 1. 이름이 `public`일 필요는 없음

### 2. 루트 디렉토리에서 실제 배포 시 접근 가능한 친구와 그렇지 않은 친구의 구분 필요

추후 CopyWebpackPlugin 등을 사용할 경우 특정 폴더에 몰아넣는 게 webpack 설정에 유리할 것 같아요.
남들이 보기에도 뭐가 배포에 사용되는 파일인지 알아보기 쉬워요

### 3. 코드 내의 외부 url을 통한 파일 요청에 대해 개발 서버와 배포 서버에서의 처리가 다름

개발 서버에서는 `/src/**/*`으로 접근이 가능하지만 배포 서버는 안돼요
두 서버의 간극을 줄이는 게 원활한 배포로 나아가는 길이라 생각합니다.

### 4. 사람을 믿을 수 없음

루트 디렉토리에 몰아넣어도 저희는 다 아니까 그냥 하면 되는데 다른 사람이 중간에 들어온다면 이건 배포에 쓰고 안쓰고 인수인계를 해줘야 그분이 코드를 이상하게 안 짜요

**Comment by hozzijeong** [comment]
제일 마지막 이유가 무엇보다 가슴에 와닿네요

**Comment by bassyu** [comment]
2번도 마음에 드는 근거네요! 좋습니다.

**Comment by WaiNaat** [comment]
원만한 합의 하에 `public` 만들기로 했어요~~
